docs/mt-config.md

```diff
@@ -118,11 +143,192 @@ The currently supported values for `<type>` are:
+      incoming and target.
+- Interleave new commits from all dirs
+- Generate merge commit between:
+    - final interleaved commit (first-parent) and
+    - monorepo equivalents of already-translated HEADs.
+- Update branch head to merge commit
+
+### Tag
+
+Follow steps for creating a new branch, but make a tag instead.
+
+
+### Rough sketch of v3 of the algorithm
+
+This moves the listing of commits directly into the C++ program to allow it to
+get incremental results and use processing of normal dirs to affect how repeats
+are done.  (As of time of writing, the algorithm has shifted from the below,
+but it's fairly close to what's implemented.)
+
+ 1. Find the dir commits to interleave.  Ignore repeats until later.
+    - For each dir:
+        - If it has a start commit:
+            - If not ancestor of goal, back up to the merge base.
+            - List "first" side of ancestry path (equivalent to first parents,
+              if that's in the ancestry path).
+        - Else:
+            - Look at (all) first parents.
+    - Note each commit's timestamp, fudging it if necessary to be monotonically
+      non-decreasing.
+ 2. Mark which commits have been translated.
+    - For each dir:
+        - Do a binary search to find where the transition happens.
+ 3. Sort by commit timestamp, interleaving commits (no translation yet).
+    - Stable sort will respect topological order because of commit timestamp
+      fudging above.
+ 4. Fast-forward through the interleaved commits to find the first commit to
+    translate, refining the start commits.
+    - Keep track of the last commit for each dir before finding the first
+      interleaved commit to translate.
+        - This becomes the start commit for that dir (even if that dir has more
+          already-translated commits).
+        - Note whether it's a change from the original start commit.
+    - Check for commit timestamp tie-breakers that are already translated,
+      and reorder locally as if we had sorted this way above.
+ 5. Convert start commits to monorepo commits.
+ 6. Find the repeat commits to interleave.
+    - If start is not an ancestor of the goal, discard it.
+    - List first parents of the goal, excluding all monorepo start commits,
+      and excluding any commits older than monorepo start commits.
+    - Filter out commits that don't modify repeated paths.
+ 7. Sort repeat commits, interleaving with other commits by commit timestamp.
+ 8. Determine whether to discard HEAD.
+    - Discard if there are no start commits.
+    - Discard if a start commit had to back up to merge base.
+    - Discard if repeat start commit was discarded.
+    - Discard if any content in HEAD is wrong vs. start commits.
+ 9. Discard repeat start commit if HEAD was discarded.
+10. Find a repeat start commit if necessary.
+    - List the newest first parent commit of the oldest repeat commit being
+      interleaved that modifies a repeated path.
```

Comment:
"Discard repeat start commit" coming before "find a repeat start commit" is a little confusing. Would it make more sense to just have the "find if necessary" and clarify that it's not necessary if HEAD was discarded?